test/test_ssl.rb

```diff
@@ -1222,6 +1222,59 @@ def test_get_ephemeral_key
     end
   end
 
+  def test_fallback_scsv
+    pend "Fallback SCSV is not supported" unless OpenSSL::SSL::SSLContext.method_defined?( :enable_fallback_scsv)
+
+    start_server do |port|
+      ctx = OpenSSL::SSL::SSLContext.new
+      ctx.ssl_version = :TLSv1_2
```

Comment:
Please use #max_version= for client SSL contexts, too.

Reply:
Ok, fixed.
But for client, it's strange to have to use min/max, because AFAIK, SSLContext doesn't handle fallback by itself and support only a single version (the max_version).

Reply:
Yes, in a real use case, the client will surely enable only one protocol version. The point is, in this case, that #ssl_version= should not be used because it's an old and deprecated method. New applications should write as:

    ctx.min_version = ctx.max_version = ...

Reply:
In this case, why not reimplement old ssl_version with min/max_version under the hood ?
min/max is cool for server part, but for client part, better to have a single version, no ?

Reply:
Actually It's been rewritten to call #min_version= and #max_version=. But it's a different story whether we should recommend using that or not.

https://github.com/ruby/openssl/blob/d834e8614b9847c442c4ccd2cd7db322aa25a0d1/lib/openssl/ssl.rb#L186

It's a decision of the OpenSSL C API that the SSL/TLS protocol version specific SSL methods, such as TLSv1_2_method() which corresponds to :TLSv1_2, are now deprecated. Please see TLSv1_2_method(3).

https://www.openssl.org/docs/manmaster/man3/TLSv1_2_method.html
(The web site seems down now)